### PR TITLE
fix: Missing errorReports when using DELETE strategy and validation fails - [ DHIS2-11300 ] - 2.38

### PR DESCRIPTION
**Pull request by lucaCambi77**
https://jira.dhis2.org/browse/DHIS2-11300
adds PreCheckMandatoryFieldsValidationHook for DELETE strategy

### CODE REVIEW

File: dhis-2/dhis-services/dhis-service-tracker/src/test/java/org/hisp/dhis/tracker/validation/EnrollmentImportValidationTest.java [hunk 1]
```diff
@@ -91,6 +91,25 @@ void testEnrollmentValidationOkAll()
...
+        TrackerImportReport trackerImportReport = trackerImportService.importTracker( params );
+        assertEquals( 0, trackerImportReport.getValidationReport().getErrors().size() );
+        assertEquals( TrackerStatus.OK, trackerImportReport.getStatus() );
+
+        params = createBundleFromJson( "tracker/validations/enrollments_te_missing_ou.json" );
+        params.setImportStrategy( TrackerImportStrategy.DELETE );
+        trackerImportReport = trackerImportService.importTracker( params );
+        assertEquals( 1, trackerImportReport.getValidationReport().getErrors().size() );
+        assertThat( trackerImportReport.getValidationReport().getErrors(),
+            hasItem( hasProperty( "errorCode", equalTo( TrackerErrorCode.E1122 ) ) ) );
```
Comment (teleivo) [inline dhis-2/dhis-services/dhis-service-tracker/src/test/java/org/hisp/dhis/tracker/validation/EnrollmentImportValidationTest.java:110]: you should be able to use

`assertTrue( trackerImportReport.getValidationReport().hasError( err -> TrackerErrorCode.E1122 == err.getErrorCode()))`

we added `hasError` to the report to help with testing if a certain error is in there 😄

File: dhis-2/dhis-services/dhis-service-tracker/src/test/java/org/hisp/dhis/tracker/validation/EnrollmentImportValidationTest.java [hunk 2]
```diff
@@ -91,6 +91,25 @@ void testEnrollmentValidationOkAll()
...
 
+    @Test
+    void testEnrollmentValidationFailMissingOrgUnitDeleteStrategy()
+        throws IOException
+    {
+
+        TrackerImportParams params = createBundleFromJson( "tracker/validations/enrollments_te_enrollments-data.json" );
+        params.setImportStrategy( TrackerImportStrategy.CREATE );
+        TrackerImportReport trackerImportReport = trackerImportService.importTracker( params );
+        assertEquals( 0, trackerImportReport.getValidationReport().getErrors().size() );
```
Comment (teleivo) [inline dhis-2/dhis-services/dhis-service-tracker/src/test/java/org/hisp/dhis/tracker/validation/EnrollmentImportValidationTest.java:102]: can you please replace it with `assertFalse(trackerImportReport.getValidationReport().hasErrors())`

File: dhis-2/dhis-services/dhis-service-tracker/src/test/java/org/hisp/dhis/tracker/validation/EnrollmentImportValidationTest.java [hunk 3]
```diff
@@ -91,6 +91,25 @@ void testEnrollmentValidationOkAll()
...
+        assertEquals( TrackerStatus.OK, trackerImportReport.getStatus() );
+
+        params = createBundleFromJson( "tracker/validations/enrollments_te_missing_ou.json" );
+        params.setImportStrategy( TrackerImportStrategy.DELETE );
+        trackerImportReport = trackerImportService.importTracker( params );
```
Comment (teleivo) [inline dhis-2/dhis-services/dhis-service-tracker/src/test/java/org/hisp/dhis/tracker/validation/EnrollmentImportValidationTest.java:107]: can you please add newlines around the 

`trackerImportReport = trackerImportService.importTracker( params );`

so its clear on first sight what method we are testing.

**Review by teleivo — APPROVED** [review]
some minor comments 😄  👍🏻

File: dhis-2/dhis-services/dhis-service-tracker/src/test/resources/tracker/validations/enrollments_te_missing_ou.json [hunk 4]
```diff
@@ -0,0 +1,19 @@
+{
```
Comment (enricocolasante) [inline dhis-2/dhis-services/dhis-service-tracker/src/test/resources/tracker/validations/enrollments_te_missing_ou.json:1]: The idea of delete payload was to have the minimal information to in it
Something like `{
  "enrollments": [
    {
       "enrollment": "MNWZ6hnuhSw"
    }
  ]
}`

@stian-sandvold @Bekkalizer Can you confirm or disprove this?

Comment (lucaCambi77) [reply dhis-2/dhis-services/dhis-service-tracker/src/test/resources/tracker/validations/enrollments_te_missing_ou.json:1]: yes that could be the idea but the validation fails, so we need either to make it even with other strategies or modify the validation

Comment (teleivo) [reply dhis-2/dhis-services/dhis-service-tracker/src/test/resources/tracker/validations/enrollments_te_missing_ou.json:1]: good point @enricocolasante glad you spotted this 😬

Comment (stian-sandvold) [reply dhis-2/dhis-services/dhis-service-tracker/src/test/resources/tracker/validations/enrollments_te_missing_ou.json:1]: Hey, the idea is that you just need the authority to delete, and delete something that exists without any unbreakable references to perform a delete. The payload @enricocolasante posted should be all you need, then all the other required info should be fetched before validation, and only the minimum should be validated.

**Pull request closed without merging**